as-controller-board-status/main.go

```diff
@@ -18,66 +17,68 @@ const (
 )
 
 func main() {
-	// Create an instance of the EdgeX SDK and initialize it
-	edgexSdk := &appsdk.AppFunctionsSDK{ServiceKey: serviceKey}
-	err := edgexSdk.Initialize()
-	if err != nil {
-		edgexSdk.LoggingClient.Error(fmt.Sprintf("SDK initialization failed: %v", err.Error()))
-		os.Exit(-1)
+	// See https://docs.edgexfoundry.org/2.2/microservices/application/ApplicationServices/
+	//       for documentation on application services.
+	service, ok := pkg.NewAppService(serviceKey)
+	if !ok {
+		os.Exit(1)
 	}
 
+	lc := service.LoggingClient()
+
 	// Get the application's settings from the configuration.toml file
-	appSettings := edgexSdk.ApplicationSettings()
+	appSettings := service.ApplicationSettings()
 	if appSettings == nil {
-		edgexSdk.LoggingClient.Error("No application settings found")
-		os.Exit(-1)
+		lc.Error("No application settings found")
+		os.Exit(1)
 	}
 
 	boardStatus := functions.CheckBoardStatus{
 		DoorClosed:    true, // Set default door state to closed
 		Configuration: new(functions.ControllerBoardStatusAppSettings),
+		Service:       service,
 	}
 
 	// Retrieve & parse the required application settings into a proper
 	// configuration struct
-	err = utilities.MarshalSettings(appSettings, boardStatus.Configuration, false)
+	err := utilities.MarshalSettings(appSettings, boardStatus.Configuration, false)
 	if err != nil {
-		edgexSdk.LoggingClient.Error(fmt.Sprintf("Application settings could not be processed: %v", err.Error()))
-		os.Exit(-1)
+		lc.Errorf("Application settings could not be processed: %v", err.Error())
+		os.Exit(1)
 	}
 
 	boardStatus.MaxTemperatureThreshold = boardStatus.Configuration.MaxTemperatureThreshold
 	boardStatus.MinTemperatureThreshold = boardStatus.Configuration.MinTemperatureThreshold
 
 	// Create the function pipeline to run when an event is read on the device channels
-	err = edgexSdk.SetFunctionsPipeline(
-		transforms.NewFilter([]string{boardStatus.Configuration.DeviceName}).FilterByDeviceName,
+	err = service.SetFunctionsPipeline(
+		transforms.NewFilterFor([]string{boardStatus.Configuration.DeviceName}).FilterByDeviceName,
 		boardStatus.CheckControllerBoardStatus,
 	)
 	if err != nil {
-		edgexSdk.LoggingClient.Error(fmt.Sprintf("SDK initialization failed: %v", err.Error()))
-		os.Exit(-1)
+		lc.Errorf("SDK initialization failed: %v", err.Error())
+		os.Exit(1)
 	}
 
 	// Subscribe to the EdgeX notification service
 	err = boardStatus.SubscribeToNotificationService()
 	if err != nil {
-		edgexSdk.LoggingClient.Info(fmt.Sprintf("Error subscribing to edgex notification service %s", err.Error()))
-		os.Exit(-1)
+		lc.Errorf("Error subscribing to EdgeX notification service %s", err.Error())
+		os.Exit(1)
 	}
 
 	// Add the "status" REST API route
-	err = edgexSdk.AddRoute("/status", functions.GetStatus, "GET", "OPTIONS")
+	err = service.AddRoute("/status", functions.GetStatus, "GET", "OPTIONS")
```

Comment:
Wrap in a Controller . Allows for logging.

Reply:
yap, next PR